VCFv4.3.tex

```diff
@@ -406,12 +406,14 @@ \subsubsection{Genotype fields}
       GQ		& 1		& Integer	& Conditional genotype quality \\
       GT		& 1		& String	& Genotype \\
       HQ		& 2		& Integer	& Haplotype quality \\
+      LAA		& . 		& Integer	& Local Alternate Alleles the 1-based index into the alternate alleles indicating which are relevant for the current sample.\\
```

Comment:
```suggestion
      LAA		& . 		& Integer	& Local Alternate Alleles the strictly-increasing, 1-based indices into the alternate alleles indicating which are relevant for the current sample.\\
```

Reply:
Or maybe:
```suggestion
      LAA		& . 		& Integer	& 1-based index indicating which alternate alleles are relevant (local) for the current sample\\
```

Reply:
SGTM but let's say the indices have to be strictly increasing; obviously that's a pedantic constraint but without it, the ordering of LPL could be unclear as well.

Reply:
Suggestion updated.

Reply:
Dwelling on this further -- @yfarjoun @lbergelson -- I'm worried just slightly about confusion over the LPL ordering. Lets say the joint VCF has REF=A ALT=G,C,T and the sample GVCF has ALT=T,G. so LPL=1,3 if we require the indices to be in order, but this order is flipped with respect to the GVCF.

The merging algo then has to be really careful to [permute the GVCF PL values](https://github.com/dnanexus-rnd/GLnexus/blob/237e9bcbcd685445ab0aa65ecbc6c29cefe1b26e/src/diploid.cc#L16-L43) in just the right way to match that. In the wild, I'm nervous this is going to lead to difficult-to-detect bugs with subtle swapping of probability values. Did you have any experience with this in developing the idea? Should we permit the merger to write an out-of-order LAA (this might just shift the potential confusion to readers)? I don't have a better idea right now, it just seems to me like a significant pitfall at least warranting explication where LPL is discussed.

Reply:
I agree that this is a concern.  The only actual implementation so far of a format with local alleles decided to store the allele string itself instead of the index because they didn't want to deal with the reordering problem.  So unordered [T, G] instead of [1, 3]. Move the difficulties from the merge step to the reading step in that now if you want to look up non-local values associated with an allele you have to do a search in the INFO allele values to find the corresponding index.  I

My thinking was that it was better to require the indexes as numbers and do the reordering, but I think there's a strong argument the other way.

Reply:
Thanks, I'll spawn a background thread on this. BTW, if anyone knows a better formula than the following to go from PL index `gt` to the indices `(i,j)` of the constituent alleles, I would like to know. This is seriously the best I could come up with, holy sh**.

```
    unsigned j = (unsigned)((sqrt(8*gt+1)-1.0)/2.0);
    unsigned i = gt - j*(j+1)/2;
```

Reply:
I like @cyenyxe's suggestion, it solves a possible confusion. It also allows to determine the LPL ordering per allele-set, samples with the same set of local alleles are guaranteed to use the same order.

@mlin Note that the proposed formula is for the special case of diploid genotypes. In my experience, in practice it is sufficient to pre-compute the indexes using nested/recursive loops as suggested in the description of GL, section 1.6.2.

Reply:
why do the indexes need to be strictly increasing? 

One of the main points is to lessen the burden on the creation of these pVCFs...so that when the alleles change the only change that needs to happen in the format field is the LAA. if we require strictly increasing order, that goes out the window.

Reply:
If LAA are strictly increasing, LPL's are guaranteed to have the same order as 1) the original PLs and 2) samples with the same set of alleles.

Editing of VCFs is much less frequent operation than read-only processing, therefore it is more important to optimize the speed and convenience of the latter.